libqtile/state.py

```diff
@@ -45,6 +53,39 @@ def apply(self, qtile):
             Rearrange the windows in the specified Qtile object according to
             this QtileState.
         """
+        try:
+            for group in qtile.groups:
+                x = []
+                for i in self.focusHistory[group.name]:
+                    x.append(qtile.windowMap[i.wid])
+                group.focusHistory = x
```

Comment:
Yes, that a nice suggestion .I will add it to the changes. Thanks :)

I add that in `line 45` in `__init__` method